File: ws/acuña/u2/WS23UnitTest/ComplexOperations/src/test/java/ec/edu/espe/complexoperations/controler/OperationTest.java
package ec.edu.espe.complexoperations.controler;


import static org.junit.jupiter.api.Assertions.*;

/**
 *
 * @author ChelooWs
 */
public class OperationTest {
    
    public OperationTest() {
    }

    @org.junit.jupiter.api.BeforeAll
    public static void setUpClass() throws Exception {
    }

    @org.junit.jupiter.api.AfterAll
    public static void tearDownClass() throws Exception {
    }

    @org.junit.jupiter.api.BeforeEach
    public void setUp() throws Exception {
    }

    @org.junit.jupiter.api.AfterEach
    public void tearDown() throws Exception {
    }
    


    /**
     * Test of addTwoNumbers method, of class Operation.
     */
    @org.junit.jupiter.api.Test
    public void testAddTwoNumbers() {
        System.out.println("addTwoNumbers");
        float addend1 = 1;
        float addend2 = 2;
        float expResult = 3.0F;
        float result = Operation.addTwoNumbers(addend1, addend2);
        assertEquals(expResult, result, 0);
      }
    public void testAddTwoNumbersOneNegative() {
        System.out.println("addTwoNumbers");
        float addend1 = -1.0F;
        float addend2 = 2.0F;
        float expResult = 3.0F;
        float result = Operation.addTwoNumbers(addend1, addend2);
        assertEquals(expResult, result, 0);
        fail("The test case is a prototype.");      
        

    }
    
}

File: hw/bonifaz/u2/HW17TestCaseSolution/ComplexOperationsJS_corretion/tests/Operation.test.js
const Operation = require('../ec.edu.espe.complexoperations.controler/Operation');

test('addTwoNumbers', () => {
    const addend1 = 1;
    const addend2 = 2;
    const expResult = 3.0;
    const result = Operation.addTwoNumbers(addend1, addend2);
    expect(result).toBeCloseTo(expResult, 1);
});

test('addTwoNumbersOneNegative', () => {
    const addend1 = 1.8;
    const addend2 = -2.4;
    const expResult = -0.6;
    const result = Operation.addTwoNumbers(addend1, addend2);
    expect(result).toBeCloseTo(expResult, 1); 
});
